### PR TITLE
Get rid of LB backend weights

### PR DESCRIPTION
**Pull request by brb**
The LB backend weights haven't been used, so let's lighten the LB code by removing all code supporting it.

```release-note/minor
Removed non-used "weight" field from "backend-address" API object.
"cilium service update" no longer accepts non-used backend weight parameters.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9254)

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

File: bpf/lib/common.h [hunk 1]
```diff
@@ -546,8 +546,7 @@ struct lb6_service_v2 {
 	__u32 backend_id;
 	__u16 count;
 	__u16 rev_nat_index;
-	__u16 weight;
-	__u16 pad;
+	__u32 pad;
```
Comment (ianvernon) [inline bpf/lib/common.h:549]: Any concerns about struct alignments here on upgrade? This isn't my area of expertise, so I may be worrying for no reason.

**Review by ianvernon — APPROVED** [review]

**Review by aanm — APPROVED** [review]

File: bpf/lib/common.h (hunk 1 above)
Comment (aanm) [reply bpf/lib/common.h:549]: +1

Comment (brb) [reply bpf/lib/common.h:549]: Fair concern, but the alignment of the fields doesn't change. Otherwise, the alignchecker would have reported an error.

**Comment by brb** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

**Pull request merged**